backfill/src/main.rs

```diff
@@ -86,30 +98,52 @@ impl<Node: FullNodeComponents> BackfillExEx<Node> {
     }
 
     /// Backfills the given range of blocks in parallel, calling the
-    /// [`Self::process_committed_chain`] method for each block.
-    async fn backfill(&self, range: RangeInclusive<BlockNumber>) -> eyre::Result<()> {
-        self.backfill_job_factory
+    /// [`Self::process_committed_chain`] method for each block. Requires acquiring a semaphore
+    /// permit that limits the number of concurrent backfills. The backfill job is spawned in a
+    /// separate task.
+    ///
+    /// Returns an error if the semaphore permit could not be acquired.
+    fn backfill(&self, range: RangeInclusive<BlockNumber>) -> eyre::Result<()> {
+        let permit = self
+            .backfill_semaphore
+            .clone()
+            .try_acquire_owned()
+            .map_err(|err| eyre::eyre!("concurrent backfills limit reached: {err:?}"))?;
```

Comment:
this needs to be moved into spawn and await acquire

Reply:
it will just block the RPC request, right? I explicitly return an error if we ran out of permits.

Reply:
I see